libs/wingsdk/src/http/http.ts

```diff
@@ -0,0 +1,259 @@
+   * Otherwise the resource will be downloaded from the server and the cache will be updated.
+   * If there is no match, the browser will make a normal request, and will update the cache with the downloaded resource.
+   */
+  DEFAULT = "default",
+  /**
+   * The browser fetches the resource from the remote server without first looking in the cache,
+   * and will not update the cache with the downloaded resource.
+   */
+  NO_STORE = "no-store",
+  /**
+   * The browser fetches the resource from the remote server without first looking in the cache,
+   * but then will update the cache with the downloaded resource.
+   */
+  RELOAD = "reload",
+  /**
+   * The browser looks for a matching request in its HTTP cache.
+   * If there is a match, fresh or stale, the browser will make a conditional request to the remote server.
+   * If the server indicates that the resource has not changed, it will be returned from the cache. Otherwise the resource will be downloaded from the server and the cache will be updated.
+   * If there is no match, the browser will make a normal request, and will update the cache with the downloaded resource.
+   */
+  NO_CACHE = "no-cache",
+  /**
+   * The browser looks for a matching request in its HTTP cache.
+   * If there is a match, fresh or stale, it will be returned from the cache.
+   * If there is no match, the browser will make a normal request, and will update the cache with the downloaded resource.
+   */
+  FORCE_CACHE = "force-cache",
+}
+/**
+ * The redirect read-only property that contains the mode for how redirects are handled.
+ */
+export enum RequestRedirect {
+  /**
+   * Follow all redirects incurred when fetching a resource.
+   */
+  FOLLOW = "follow",
+  /**
+   * Return a network error when a request is met with a redirect.
+   */
+  ERROR = "error",
+}
+/**
+ * The request's method
+ */
+export enum HttpMethods {
+  /**
+   * GET
+   */
+  GET = "GET",
+  /**
+   * PUT
+   */
+  PUT = "PUT",
+  /**
+   * DELETE
+   */
+  DELETE = "DELETE",
+  /**
+   * PATCH
+   */
+  PATCH = "PATCH",
+  /**
+   * POST
+   */
+  POST = "POST",
+}
+/**
+ * An object containing any custom settings that you want to apply to the request.
+ */
+export interface RequestOptions {
+  /**
+   * The request method, e.g., GET, POST. The default is GET.
+   */
+  readonly method?: HttpMethods;
+  /**
+   * Any headers you want to add to your request.
+   */
+  readonly headers?: Record<string, string>;
+  /**
+   * Any body that you want to add to your request.
+   * Note that a request using the GET or HEAD method cannot have a body.
+   */
+  readonly body?: string;
+  /**
+   * The cache mode you want to use for the request.
+   */
+  readonly cache?: RequestCache;
+  /**
+   * he redirect mode to use: follow, error. The default is follow.
+   */
+  readonly redirect?: RequestRedirect;
+  /**
+   * A string specifying "no-referrer", client, or a URL. The default is "about:client".
```

Comment:
Is that really the default?

Reply:
it's all taken from here: 
https://developer.mozilla.org/en-US/docs/Web/API/Request/Request